### PR TITLE
Bluetooth: Update terms in public API to spec v5.3

### PR DESCRIPTION
**Pull request by carlescufi**
The Bluetooth Core Specification, version 5.3, has introduced multiple
changes to several widely-used terms in order to make them inclusive.
Update the public API to reflect this, excluding hci.h, which will be
done in a subsequent commit.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

See #38348

### CODE REVIEW

File: subsys/bluetooth/shell/bt.c [hunk 1]
```diff
@@ -2923,7 +2923,7 @@ static int cmd_wl_rem(const struct shell *shell, size_t argc, char *argv[])
 		return err;
 	}
 
-	err = bt_le_whitelist_rem(&addr);
+	err = bt_le_fa_list_rem(&addr);
 	if (err) {
 		shell_error(shell, "Remove from whitelist failed (err %d)",
 			    err);
```
Comment (LingaoM) [inline subsys/bluetooth/shell/bt.c:2929]:
```suggestion
		shell_error(shell, "Remove from filter accept list failed (err %d)",
			    err);
```

File: subsys/bluetooth/shell/bt.c [hunk 2]
```diff
@@ -2936,7 +2936,7 @@ static int cmd_wl_clear(const struct shell *shell, size_t argc, char *argv[])
 {
 	int err;
 
-	err = bt_le_whitelist_clear();
+	err = bt_le_fa_list_clear();
 	if (err) {
 		shell_error(shell, "Clearing whitelist failed (err %d)", err);
```
Comment (LingaoM) [inline subsys/bluetooth/shell/bt.c:2941]: Same too.

File: subsys/bluetooth/shell/bt.c [hunk 3]
```diff
@@ -2903,7 +2903,7 @@ static int cmd_wl_add(const struct shell *shell, size_t argc, char *argv[])
 		return err;
 	}
 
-	err = bt_le_whitelist_add(&addr);
+	err = bt_le_fa_list_add(&addr);
 	if (err) {
 		shell_error(shell, "Add to whitelist failed (err %d)", err);
```
Comment (LingaoM) [inline subsys/bluetooth/shell/bt.c:2908]: Also.

File: subsys/bluetooth/shell/bt.c [hunk 4]
```diff
@@ -2440,9 +2440,9 @@ static int cmd_bonds(const struct shell *shell, size_t argc, char *argv[])
 static const char *role_str(uint8_t role)
 {
 	switch (role) {
-	case BT_CONN_ROLE_MASTER:
+	case BT_CONN_ROLE_CENTRAL:
 		return "Master";
-	case BT_CONN_ROLE_SLAVE:
+	case BT_CONN_ROLE_PERIPH:
 		return "Slave";
```
Comment (LingaoM) [inline subsys/bluetooth/shell/bt.c:2446]: +1

File: subsys/bluetooth/host/scan.c [hunk 5]
```diff
@@ -991,7 +991,7 @@ int bt_le_scan_start(const struct bt_le_scan_param *param, bt_le_scan_cb_t cb)
 
 #if defined(CONFIG_BT_WHITELIST)
```
Comment (LingaoM) [inline subsys/bluetooth/host/scan.c:992]: It seems that this thing is tricky.

Comment (carlescufi) [reply subsys/bluetooth/host/scan.c:992]: Ops, I need to change that, thanks!

File: include/bluetooth/conn.h [hunk 6]
```diff
@@ -303,10 +303,13 @@ struct bt_conn_br_info {
...
-/** Connection role (master or slave) */
+/** Connection role (central or peripheral) */
+#define BT_CONN_ROLE_MASTER __DEPRECATED_MACRO 0
+#define BT_CONN_ROLE_SLAVE __DEPRECATED_MACRO 1
+
 enum {
-	BT_CONN_ROLE_MASTER,
-	BT_CONN_ROLE_SLAVE,
+	BT_CONN_ROLE_CENTRAL = 0,
+	BT_CONN_ROLE_PERIPH = 1,
```
Comment (Thalley) [inline include/bluetooth/conn.h:312]:
```suggestion
	BT_CONN_ROLE_PERIPHERAL = 1,
```

I think it's worth using the full name here

File: subsys/bluetooth/shell/bt.c [hunk 7]
```diff
@@ -1985,8 +1985,8 @@ static const char *get_conn_type_str(uint8_t type)
 static const char *get_conn_role_str(uint8_t role)
 {
 	switch (role) {
-	case BT_CONN_ROLE_MASTER: return "master";
-	case BT_CONN_ROLE_SLAVE: return "slave";
+	case BT_CONN_ROLE_CENTRAL: return "master";
+	case BT_CONN_ROLE_PERIPH: return "slave";
```
Comment (Thalley) [inline subsys/bluetooth/shell/bt.c:1989]:
```suggestion
	case BT_CONN_ROLE_PERIPH: return "peripheral";
```

File: subsys/bluetooth/shell/bt.c (hunk 4 above)
Comment (Thalley) [inline subsys/bluetooth/shell/bt.c:2446]:
```suggestion
	case BT_CONN_ROLE_PERIPH:
		return "Peripheral";
```

**Review by Thalley — CHANGES_REQUESTED** [review]
What about include/bluetooth/hci.h? Is that fixed by another PR?

**Comment by carlescufi** [comment]
@LingaoM thanks for the quick review! I've addressed all your comments

File: include/bluetooth/conn.h (hunk 6 above)
Comment (carlescufi) [reply include/bluetooth/conn.h:312]: I was expecting this :)
No big preference for me here, but let's run a quick poll @cvinayak @joerchan @alwa-nordic @asbjornsabo @Vudentz @jhedberg @LingaoM @trond-snekvik @thoh-ot @kruithofa

**Comment by carlescufi** [comment]
> What about include/bluetooth/hci.h? Is that fixed by another PR?

Yes, that will be separate because hci.h is not quite technically part of the public API, so I want to deal with that later.

File: subsys/bluetooth/host/scan.c [hunk 8]
```diff
@@ -209,7 +209,7 @@ static int start_le_scan_legacy(uint8_t scan_type, uint16_t interval, uint16_t w
 	set_param.interval = sys_cpu_to_le16(interval);
 	set_param.window = sys_cpu_to_le16(window);
 
-	if (IS_ENABLED(CONFIG_BT_WHITELIST) &&
+	if (IS_ENABLED(CONFIG_BT_FA_LIST) &&
 	    atomic_test_bit(bt_dev.flags, BT_DEV_SCAN_WL)) {
```
Comment (LingaoM) [inline subsys/bluetooth/host/scan.c:213]:
```suggestion
	    atomic_test_bit(bt_dev.flags, BT_DEV_SCAN_FL)) {
```

**Comment by carlescufi** [comment]
@Thalley thanks for the quick review. I've addressed most of your comments.

File: subsys/bluetooth/shell/bt.c [hunk 9]
```diff
@@ -2891,7 +2891,7 @@ static int cmd_auth_pairing_confirm(const struct shell *shell,
 	return 0;
 }
 
-#if defined(CONFIG_BT_WHITELIST)
+#if defined(CONFIG_BT_FA_LIST)
 static int cmd_wl_add(const struct shell *shell, size_t argc, char *argv[])
```
Comment (LingaoM) [inline subsys/bluetooth/shell/bt.c:2895]:
```suggestion
static int cmd_fl_add(const struct shell *shell, size_t argc, char *argv[])
```

File: include/bluetooth/conn.h (hunk 6 above)
Comment (asbjornsabo) [reply include/bluetooth/conn.h:312]: If we use the full name for the central, we can do so for the peripheral as well.

File: subsys/bluetooth/host/scan.c (hunk 8 above)
Comment (asbjornsabo) [reply subsys/bluetooth/host/scan.c:213]: Agree

**Review by asbjornsabo — APPROVED** [review]
I am in favour of changing PERIPH to PERIPHERAL, and the SACN_WL to SCAN_FL
Otherwise LGTM.

File: subsys/bluetooth/host/scan.c (hunk 8 above)
Comment (carlescufi) [reply subsys/bluetooth/host/scan.c:213]: This PR only changes the public API @LingaoM. We will need further cleanups, but I want to keep those separate.

File: subsys/bluetooth/shell/bt.c (hunk 9 above)
Comment (carlescufi) [reply subsys/bluetooth/shell/bt.c:2895]: This PR only changes the public API @LingaoM. We will need further cleanups, but I want to keep those separate.

File: include/bluetooth/conn.h (hunk 6 above)
Comment (jhedberg) [reply include/bluetooth/conn.h:312]: I'd use the full name to be consistent with "central"

Comment (cvinayak) [reply include/bluetooth/conn.h:312]: Peripheral it is.

BT Core TS uses "CEN" and "PER" abbreviations, but Central and Peripheral is better in API

File: subsys/bluetooth/host/Kconfig [hunk 10]
```diff
@@ -173,19 +173,23 @@ config BT_SETTINGS_USE_PRINTK
 endif # BT_SETTINGS
 
 config BT_WHITELIST
-	bool "Enable whitelist support"
+	bool "DEPRECATED: Use BT_FA_LIST instead"
```
Comment (joerchan) [inline subsys/bluetooth/host/Kconfig:176]: Is this a new policy? I cannot remember seeing deprecation of Kconfig options.

File: include/bluetooth/bluetooth.h [hunk 11]
```diff
@@ -1852,50 +1855,65 @@ void bt_le_scan_cb_register(struct bt_le_scan_cb *cb);
...
  *
  * @return Zero on success or error code otherwise, positive in case of
  *         protocol error or negative (POSIX) in case of stack internal error.
  */
-int bt_le_whitelist_rem(const bt_addr_le_t *addr);
+int bt_le_fa_list_rem(const bt_addr_le_t *addr);
+__deprecated
+static inline int bt_le_whitelist_rem(const bt_addr_le_t *addr)
+{
+	return bt_le_fa_list_rem(addr);
```
Comment (joerchan) [inline include/bluetooth/bluetooth.h:1896]: I would take this opportunity to rename from `_rem` to `_remove` as well. Aligns up with period adv list for example.

File: include/bluetooth/bluetooth.h [hunk 12]
```diff
@@ -1852,50 +1855,65 @@ void bt_le_scan_cb_register(struct bt_le_scan_cb *cb);
...
+ * the filter accept list, i.e advertiser or scanner using a filter accept list
+ * or automatic connecting to devices using filter accept list.
  *
  * @param addr Bluetooth LE identity address.
  *
  * @return Zero on success or error code otherwise, positive in case of
  *         protocol error or negative (POSIX) in case of stack internal error.
  */
-int bt_le_whitelist_rem(const bt_addr_le_t *addr);
+int bt_le_fa_list_rem(const bt_addr_le_t *addr);
```
Comment (joerchan) [inline include/bluetooth/bluetooth.h:1892]: Using mixed abbreviation of `fa` and `FILTER_ACCEPT_LIST` in scan options, and `FILTER` in other places.

`bt_le_filter_accept_list_add` and friends does not seem so bad to me, so I would suggest that instead.

File: subsys/bluetooth/host/Kconfig (hunk 10 above)
Comment (carlescufi) [reply subsys/bluetooth/host/Kconfig:176]: We've done a mix of both in the past, I'm taking the full path here.

File: include/bluetooth/bluetooth.h (hunk 12 above)
Comment (carlescufi) [reply include/bluetooth/bluetooth.h:1892]: OK, fair enough. `bt_le_filter_accept_list_remove` is going to be a mouthful though ;)

File: include/bluetooth/conn.h (hunk 6 above)
Comment (carlescufi) [reply include/bluetooth/conn.h]: Thanks for all the feedback. I have replaced it with the full word.

**Comment by carlescufi** [comment]
@LingaoM @Thalley @cvinayak @joerchan @alwa-nordic can you approve if happy please? I'd like to merge this before today's TSC.

**Review by LingaoM — APPROVED** [review]

File: subsys/bluetooth/shell/bt.c [hunk 13]
```diff
@@ -904,7 +904,7 @@ static int cmd_scan(const struct shell *shell, size_t argc, char *argv[])
 		} else if (!strcmp(arg, "nodups")) {
 			options &= ~BT_LE_SCAN_OPT_FILTER_DUPLICATE;
 		} else if (!strcmp(arg, "wl")) {
-			options |= BT_LE_SCAN_OPT_FILTER_WHITELIST;
+			options |= BT_LE_SCAN_OPT_FILTER_ACCEPT_LIST;
```
Comment (Thalley) [inline subsys/bluetooth/shell/bt.c:907]: I guess we don't count the shell as public API, right? Otherwise this should also be updated.

**Review by Thalley — APPROVED** [review]
I approve the changes. I haven't verified if any renamings are missing. One small comment w.r.t. the shell, but I think that can be postponed.

File: subsys/bluetooth/shell/bt.c (hunk 13 above)
Comment (carlescufi) [reply subsys/bluetooth/shell/bt.c:907]: Right, there is another PR coming for the internal implementation. These changes here are present to eliminate any warnings related to using deprecated macros.

**Review by joerchan — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**